src/handler/http/request/dashboards/mod.rs

```diff
@@ -13,16 +13,38 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-use std::{collections::HashMap, io::Error};
+use std::collections::HashMap;
 
 use actix_web::{delete, get, http, post, put, web, HttpRequest, HttpResponse, Responder};
-use config::meta::dashboards::{ListDashboardsParams, MoveDashboard};
+use config::{
+    meta::dashboards::{v1, v2, v3, v4, v5, Dashboard, ListDashboardsParams, MoveDashboard},
+    utils::json,
+};
 use serde::Deserialize;
 
-use crate::{common::meta::http::HttpResponse as MetaHttpResponse, service::dashboards};
+use crate::{
+    common::meta::http::HttpResponse as MetaHttpResponse,
+    service::dashboards::{self, DashboardError},
+};
 
 pub mod reports;
 
+impl From<DashboardError> for HttpResponse {
```

Comment:
All logic for transforming errors that occur in the Service Layer into HTTP responses is consolidated here.